sigle/src/modules/editor/BubbleMenuItems.tsx

```diff
@@ -0,0 +1,195 @@
+  fontSize: '14px',
+  lineHeight: '18px',
+  width: '100%',
+  pl: '$2',
+  pr: '$1',
+  backgroundColor: 'transparent',
+  outline: 'none',
+});
+
+const BubbleMenuButton = styled('button', {
+  color: '$gray12',
+
+  '@xl': {
+    color: '$gray1',
+  },
+
+  variants: {
+    active: {
+      true: {
+        color: '$orange10',
+      },
+    },
+  },
+});
+
+interface ResetLink {
+  resetLink: () => void;
+}
+
+interface BubbleMenuItemsProps {
+  editor: Editor;
+  iconSize?: number;
+}
+
+export const BubbleMenuItems = forwardRef<ResetLink, BubbleMenuItemsProps>(
+  ({ editor, iconSize = 18 }: BubbleMenuItemsProps, ref) => {
+    const [linkState, setLinkState] = useState({ open: false, value: '' });
+
+    useImperativeHandle(ref, () => ({
+      resetLink,
+    }));
+
+    // Listen to any key press to detect cmd + k and activate the link edition
+    useEffect(() => {
+      function onKeyDown(event: KeyboardEvent) {
+        // We want all our commands to start with the user pressing ctrl or cmd for mac users
+        if ((event.ctrlKey || event.metaKey) && event.key === 'k') {
+          onSelectLink();
+        }
+      }
+
+      window.addEventListener('keydown', onKeyDown);
+      return () => window.removeEventListener('keydown', onKeyDown);
+    }, []);
+
+    const onSelectLink = () => {
+      // Get href of selected link to pre fill the input
+      const existingHref = editor.isActive('link')
+        ? editor.getAttributes('link').href
+        : '';
+
+      setLinkState({
+        open: true,
+        value: existingHref,
+      });
+    };
+
+    const onSubmitLink = (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+
+      let safeLinkValue = linkState.value.trim();
+
+      if (
+        safeLinkValue &&
+        !safeLinkValue.startsWith('http') &&
+        !safeLinkValue.startsWith('#')
+      ) {
+        safeLinkValue = `https://${linkState.value}`;
+      }
+
+      if (safeLinkValue) {
+        editor
+          .chain()
+          .focus()
+          .extendMarkRange('link')
+          .setLink({ href: safeLinkValue })
+          // Set the text selection at the end of the link selection
+          // that way user can continue to type easily
+          .setTextSelection(editor.state.selection.$to.pos)
+          // Unset link selection se when the user continues to type it won't be a link
+          .unsetLink()
+          .run();
+      } else {
+        // If input text is empty we unset the link
+        editor.chain().focus().extendMarkRange('link').unsetLink().run();
+      }
+
+      resetLink();
+    };
+
+    const onKeyDown = (event: React.KeyboardEvent) => {
+      // If user press escape we hide the link input
+      if (event.key === 'Escape') {
+        event.preventDefault();
+        resetLink();
+      }
+    };
+
+    const resetLink = () => {
+      setLinkState({
+        open: false,
+        value: '',
+      });
+    };
+
+    return (
+      <>
```

Comment:
With this one, as I was unable to access the tippy styles to update (element from the lib has style tag) them to show the bubble menu at the bottom of the screen on mobile, I decided to try another approach splitting the bubble items into a separate component which could be rendered in the tooltip on desktop, and at the bottom of the screen on mobile.